tests/test_volume.py

```diff
@@ -120,6 +130,88 @@ def testProject(self):
             #  centered along the rotation axis for multiples of pi/2.
             self.assertTrue(np.allclose(vol_along_axis, prj_along_axis))
 
+    def testRotate(self):
+        # Create a stack of rotations to test. We will rotate by multiples of pi/2 from each axis.
+        rot_mat = np.empty((12, 3, 3), dtype=self.dtype)
+        axes = ["x", "y", "z"]
+        angles = [0, np.pi / 2, np.pi, 3 * np.pi / 2]
+        pairs = list(product(axes, angles))
+        for k in range(len(pairs)):
+            rot_mat[k] = Rotation.from_euler(pairs[k][0], pairs[k][1]).as_matrix()
+
+        # Rotate the basis volume (contains a 1 on each positive axis, zero elsewhere) by rotation matrices rot_mat.
+        # Nyquist frequencies are not set to zero for even resolution to improve error.
+        rot_vols = self.vols_3.rotate(0, rot_mat, nyquist=False)
+
+        # Create reference volumes
+        L = self.res
+        ref_vol = np.zeros((8, L, L, L))
+        index = 0
```

Comment:
This is just a remark, no need to change.  In this case I actually think I prefer your loops... 

```
 for ind, (i,j,k) in enumerate(product(range(2),repeat=3)):
...   print(ind, i,j,k)
... 
0 0 0 0
1 0 0 1
2 0 1 0
3 0 1 1
4 1 0 0
5 1 0 1
6 1 1 0
7 1 1 1
```